### DISCUSSION TITLE
Nested Condition/Multiple filters in TS Querybuilder

### DISCUSSION THREAD

**Discussion by bastien8060** [geldata/gel#6021] [Q&A]
I can't seem to find in the documentation how to set multiple filters on a query (`e.select`/`e.insert`, etc...)

I've tried few things:

An array of conditions:
```js
e.select(e.default.ListingApplication, (listing_application) => ({
  id: true,
  filter_single: [
    e.default.op(listing_application.user.account.email, "=", e.default.str(email)),
    e.default.op(listing_application.listing.id, "=", e.default.uuid(listing_id)),
    e.default.op(listing_application.status, "=", e.default.str("Pending")),
  ],
}));
 ```
 
Connecting all conditions with "AND" (errors with "unrecognized pattern")
```js
e.select(e.default.ListingApplication, (listing_application) => ({
  id: true,
  filter_single: e.default.op(
	    e.default.op(listing_application.user.account.email, "=", e.default.str(email)),
		"AND",
    	e.default.op(listing_application.listing.id, "=", e.default.uuid(listing_id)),
		"AND",
    	e.default.op(listing_application.status, "=", e.default.str("Pending")),
    )
}));
```

Nesting all conditions with many `e.op` operators, such that the "AND" operator only connects two conditions directly.
```js
e.select(e.default.ListingApplication, (listing_application) => ({
  id: true,
  filter_single: e.default.op(
	    e.default.op(listing_application.user.account.email, "=", e.default.str(email)),
		"AND",
		e.default.op(
    		e.default.op(listing_application.listing.id, "=", e.default.uuid(listing_id)),
			"AND",
    		e.default.op(listing_application.status, "=", e.default.str("Pending")),
		)
   )
}));
```

I also tried making a "composite filtering" like the first example here: https://www.edgedb.com/docs/clients/js/select#select-a-single-object to no avail since my conditions are based on properties of links

What am I missing? 😅

**Comment by bastien8060** [comment]
Worst case scenario, I'll just write it first in EdgeQL, but other than that, the QueryBuilder is great! I feel like the documentation overlooks a lot of important use cases though !

**Comment by stav** [comment]
I love the documentation. I find it full of good examples.

The composite filtering worked well for me.

@bastien8060 try using the key `filter`, not `filter_single`?
